decimal/decimal.nim

```diff
@@ -6,3 +6,93 @@
 # at your option. This file may not be copied, modified, or distributed except according to those terms.
 
 import decimal_lowlevel
+
+type 
+    DecimalType = ptr mpd_t
+
+const
+    DEFAULT_PREC = MPD_RDIGITS * 2
+    DEFAULT_EMAX = when (sizeof(int) == 8): 999999999999999999 else: 425000000
+    DEFAULT_EMIN = when (sizeof(int) == 8): -999999999999999999 else: -425000000
+
+var CTX: mpd_context_t
+var CTX_ADDR = addr CTX
+mpd_defaultcontext(CTX_ADDR)
+
+proc setPrec*(prec: int) =
+    if 0 < prec:
+        let success = mpd_qsetprec(CTX_ADDR, prec)
+        if success == 0:
+            echo "Bad prec" 
+
+proc `$`*(s: DecimalType): cstring = mpd_to_sci(s, 0)
+
+proc initDecimal*(): DecimalType = mpd_new(CTX_ADDR)
+
+proc newDecimal*(s: string): DecimalType =
+    result = mpd_new(CTX_ADDR)
+    mpd_set_string(result, s, CTX_ADDR)
+
+proc newDecimal*(s: int64): DecimalType =
+    result = mpd_new(CTX_ADDR)
+    mpd_set_i64(result, s, CTX_ADDR)
+
+proc newDecimal*(s: int32): DecimalType =
+    result = mpd_new(CTX_ADDR)
+    mpd_set_i32(result, s, CTX_ADDR)
+
+proc clone*(b: DecimalType): DecimalType =
+    result = initDecimal()
+    mpd_copy(result, b, CTX_ADDR)
+
+proc `+`*(a, b: DecimalType): DecimalType =
+    result = initDecimal()
+    mpd_add(result, a, b, CTX_ADDR)
+
+proc `+=`*(a, b: DecimalType) =
+    mpd_add(a, a, b, CTX_ADDR)
+
+proc `-`*(a, b: DecimalType): DecimalType =
+    result = initDecimal()
+    mpd_sub(result, a, b, CTX_ADDR)
+
+proc `-=`*(a, b: DecimalType) =
+    mpd_sub(a, a, b, CTX_ADDR)
+
+proc `*`*(a, b: DecimalType): DecimalType =
+    result = initDecimal()
+    mpd_mul(result, a, b, CTX_ADDR)
+
+proc `*=`*(a, b: DecimalType) =
+    mpd_mul(a, a, b, CTX_ADDR)
+
+proc `/`*(a, b: DecimalType): DecimalType =
+    result = initDecimal()
+    mpd_div(result, a, b, CTX_ADDR)
+
+proc `/=`*(a, b: DecimalType) =
+    mpd_div(a, a, b, CTX_ADDR)
+
+proc `^`*(a, b: DecimalType): DecimalType =
+    result = initDecimal()
+    mpd_pow(result, a, b, CTX_ADDR)
+
+proc divint*(a, b: DecimalType): DecimalType =
+    result = initDecimal()
+    mpd_divint(result, a, b, CTX_ADDR)
```

Comment:
It might be worthwhile to use `//` here, but it doesn't seem to be a convention in Nim as we usually overload `/`.

No preference.

Reply:
for now both can be used